### PR TITLE
ServeHTTP

### PR DESCRIPTION
**Pull request by DavidLu1997**
- Started `serveHTTP` function
- Added `isApi` flag to handler
- Started token checking

TODO:
- [ ] Generate tokens
- [x] Actually verify tokens
- [x] Make errors not 404
- [x] Make 404
- [x] Standardize errors

Yes this builds and passes gofmt

### CODE REVIEW

**Comment by nairboh** [comment]
+1 looks good

File: model/client.go [hunk 1]
```diff
@@ -7,4 +7,7 @@ const (
 	HEADER_FORWARDED_PROTO = "X-Forwarded-Proto"
 	HEADER_FORWARDED       = "X-Forwarded-For"
 	HEADER_REAL_IP         = "X-Real-IP"
+	HEADER_REQUEST_ID      = "X-Request-ID"
+	HEADER_AUTH            = "Authorization"
+	HEADER_BEARER          = "Bear"
```
Comment (Denton-L) [inline model/client.go]: Don't code drunk, please.

File: model/player.go [hunk 2]
```diff
@@ -5,17 +5,19 @@ package model
...
+
+	"golang.org/x/crypto/bcrypt"
 )
 
 const (
 	DEFAULT_LOCALE      = "en"
 	MIN_PASSWORD_LENGTH = 5
 	MIN_USERNAME_LENGTH = 4
 	MAX_USERNAME_LENGTH = 24
+	ADMIN_USERNAME      = "admin"
```
Comment (Denton-L) [inline model/player.go]: The user object should have a flag indicating whether they're an admin or not. This feels so dirty, man.

**Comment by Denton-L** [comment]
Will +1 once we no longer have a constant admin account.

**Comment by Denton-L** [comment]
👍

**Pull request merged**